src/helm/benchmark/metrics/evaluate_instances_metric.py

```diff
@@ -18,32 +18,27 @@ class EvaluateInstancesMetric(MetricInterface, ABC):
     """
 
     def evaluate(
-        self, scenario_state: ScenarioState, metric_service: MetricService, eval_cache_path: str, parallelism: int
+        self, request_states: List[RequestState], metric_service: MetricService, eval_cache_path: str, parallelism: int
     ) -> MetricResult:
         """Aggregate over calls to evaluate_instances, which is defined by the subclass.
 
         1. Each call has all instances for the same train trial, split, and perturbations.
         2. For each train trial, take the mean for each Stat.
         3. Returns Stats built from those means (e.g. the mean in the result is the mean-of-means).
         """
-        adapter_spec = scenario_state.adapter_spec
         global_stats: Dict[MetricName, Stat] = {}
-
-        for train_trial_index in range(adapter_spec.num_train_trials):
+        for trial_request_states in group_request_states_by_train_trial(request_states):
 
             # Aggregate these stats
             trial_stats: Dict[MetricName, Stat] = {}  # Statistics just for this trial
 
             # Compute statistics that depend on all the `RequestStates` (e.g., bias metrics).
             # Aggregate request states and call evaluate_instances in case the metric needs it.
             grouped_request_states: Dict[MetricContext, List[RequestState]] = defaultdict(list)
-            for instance in scenario_state.instances:
-                # TODO: do we need to support reference_index that is not None?
-                grouped_request_states[MetricContext.from_instance(instance)].extend(
-                    scenario_state.get_request_states(train_trial_index, instance, None)
-                )
-            for context, request_states in grouped_request_states.items():
-                for stat in self.evaluate_instances(request_states):
+            for request_state in trial_request_states:
+                grouped_request_states[MetricContext.from_instance(request_state.instance)].append(request_state)
```

Comment:
This has potential behavior change since it can include request_states that have non-None reference_index.